pkg/events/dependencies/manager.go

```diff
@@ -0,0 +1,175 @@
+package dependencies
+
+import "github.com/aquasecurity/tracee/pkg/events"
+
+// Manager is a management tree for the current dependencies of events.
+// As events can depend on one another, it manages their connection in the form of a tree.
+// The tree supports watcher functions for adding and removing nodes.
+// The manager is *not* thread-safe.
+type Manager struct {
+	nodes              map[events.ID]*EventNode
+	onAdd              []func(*EventNode)
+	onRemove           []func(*EventNode)
+	dependenciesGetter func(events.ID) events.Dependencies
+}
+
+func NewDependenciesManager(dependenciesGetter func(events.ID) events.Dependencies) *Manager {
+	return &Manager{
+		nodes:              make(map[events.ID]*EventNode),
+		dependenciesGetter: dependenciesGetter,
+	}
+}
+
+// SubscribeAdd adds a watcher function called upon the addition of an event to the tree.
+func (t *Manager) SubscribeAdd(onAdd func(*EventNode)) {
+	t.onAdd = append(t.onAdd, onAdd)
+}
+
+// SubscribeRemove adds a watcher function called upon the removal of an event from the tree.
+func (t *Manager) SubscribeRemove(onRemove func(*EventNode)) {
+	t.onRemove = append(t.onRemove, onRemove)
+}
+
+// GetEvent returns the dependencies of the given event.
+func (t *Manager) GetEvent(id events.ID) (*EventNode, bool) {
+	node := t.getNode(id)
+	if node == nil {
+		return nil, false
+	}
+	return node, true
+}
+
+// SelectEvent adds the given event to the management tree with default dependencies
+// and marks it as explicitly selected.
+// It also adds all the dependency events of the given event to the tree.
```

Comment:
Sorry for only replying now.
I am not sure what you mean by using "dependencies" rather than "dependency" makes it clearer. In the end its just singular vs plural form. In this case you are right that using dependencies here is clearer as we talk about multiple events, but in places that talk about one dependency I think the term "dependency" is clearer, no?

Reply:
Don't bother with it, it's just nit, since both phrases ultimately convey the idea of adding related events to a tree based on dependency.